src/setup-beam.js

```diff
@@ -483,31 +491,53 @@ async function get(url0, pageIdxs) {
   if (pageIdxs[0] === null) {
     return getPage(null)
   }
+
   return Promise.all(pageIdxs.map(getPage))
 }
 
 async function getWithMirrors(resourcePath, hexMirrors) {
   if (hexMirrors.length === 0) {
     throw new Error(`Could not fetch ${resourcePath} from any hex.pm mirror`)
   }
+
   const [hexMirror, ...hexMirrorsT] = hexMirrors
   try {
     return await get(`${hexMirror}${resourcePath}`, [null])
   } catch (err) {
     core.info(`get failed for URL ${hexMirror}${resourcePath}`)
   }
+
   return getWithMirrors(resourcePath, hexMirrorsT)
 }
 
 function maybePrependWithV(v) {
   if (isVersion(v)) {
     return `v${v.replace('v', '')}`
   }
+
   return v
 }
 
+function maybeRemoveVPrefix(ver) {
+  let ret = ver
+  if (isVersion(ver)) {
+    ret = ver.replace('v', '')
+  }
+
+  return ret
+}
+
+function xyzabVer(pref, suf) {
+  // This accounts for stuff like 6.0.2.0.1.0, as proposed by Erlang's
+  // https://www.erlang.org/doc/system_principles/versions.html
+  const dd = '\\.?(\\d+)?'
+  return new RegExp(
+    `${pref}v?(\\d+)${dd}${dd}${dd}${dd}${dd}(?:-rc\\.?\\d+)?(?:-otp-\\d+)?${suf}`,
+  )
+}
+
```

Comment:
This function, and it's callers, will probably be the "most difficult bits" to maintain, but the regexps here are pretty relaxed, too...